### PR TITLE
Vulnerabilities: Frontend - tar and ws

### LINKED ISSUE #437: Vulnerability: Frontend tar
# 🐛 Summary #

tar  <6.2.1
Severity: moderate
Denial of service while parsing a tar file due to lack of folders count validation - https://github.com/advisories/GHSA-f5x3-32g6-xq36

node_modules/tar

Dependency Tree:
Screenshot 2024-07-12 at 09 07 46

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #

## 🗣 Description ##

Bump versions of tar and ws to address vulnerabilities.

## 💭 Motivation and context ##

tar <6.2.1
Severity: moderate
Denial of service while parsing a tar file due to lack of folders count validation - https://github.com/advisories/GHSA-f5x3-32g6-xq36

ws 6.0.0 - 6.2.2 || 7.0.0 - 7.5.9 || 8.0.0 - 8.17.0
Severity: high
ws affected by a DoS when handling a request with many HTTP headers - https://github.com/advisories/GHSA-3h5v-q93c-6h6q

Closes issue #437
Closes issue #438 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by cduhn17 — APPROVED** [review]
LGTM

**Review by schmelz21 — APPROVED** [review]

**Review by hawkishpolicy — APPROVED** [review]
LGTM.

**Review by nickviola — APPROVED** [review]
LGTM

**Review by schmelz21 — APPROVED** [review]

File: frontend/package-lock.json
Comment (schmelz21) [inline frontend/package-lock.json:1]: @Matthew-Grayson - so it looks like the conflict is in node hash and in the ws version. If good with it, I can Accept Current on local, commit and set to merge. Have you retake a look at the commit. 
16225-16239

      "integrity": "sha512-+dbF1tHwZpXcbOJdVOkzLDxZP1ailvSxM6ZweXTegylPny803bFhA+vqBYw4s31NSAk4S2Qz+AKXK9a4wkdjcQ==

      "resolved": "https://registry.npmjs.org/ws/-/ws-7.5.10.tgz",
      "version": "7.5.10"

**Review by schmelz21 — APPROVED** [review]
Per our conversation and [Merge develop into 437-vulnerability-frontend-tar.](https://github.com/cisagov/XFD/pull/446/commits/2eaf6f47f08fe18e1f83e6d343841f4f34784d9f). LGTM

**Pull request merged**